### ISSUE TITLE
`mono` causes command-line send to fail

### ISSUE THREAD

**Issue by dcrossleyau** [neomutt/neomutt#4083]
## Expected Behaviour

Message is sent via the command-line as per NeoMutt [docs](https://neomutt.org/guide/reference.html#commandline).

## Actual Behaviour

Fails immediately.

## Steps to Reproduce

```
$ /usr/local/bin/neomutt -s test-commandline -- me@example.com \
  < ./test-content.txt
Segmentation fault: 11
```

I do not know about a 'coredump' file. Never tried to do such.

## How often does this happen?

- Always

## When did it start to happen?

After upgrade to `neomutt 20231023` \
Prior to that it has worked flawlessly. \
Previous working version was `neomutt 20231006`

No changes had been made to ~/.neomuttrc

## NeoMutt Version

```
NeoMutt 20231023
Copyright (C) 1996-2022 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Darwin 21.6.0 (x86_64)
ncurses: ncurses 6.4.20221231 (compiled with 6.4.20221231)
libiconv: 1.11
libidn2: 2.3.4 (compiled with 2.3.4)
GPGME: 1.22.0
OpenSSL: OpenSSL 3.1.3 19 Sep 2023
libnotmuch: 5.6.0
PCRE2: 10.42 2022-12-11
storage: tokyocabinet, lmdb
compression: zlib

Configure options: --prefix=/usr/local/Cellar/neomutt/20231023 --sysconfdir=/usr/local/etc --autocrypt --gss --disable-idn --idn2 --lmdb --nls --notmuch --pcre2 --sasl --sqlite --tokyocabinet --zlib --with-lua=/usr/local/opt/lua --with-ncurses=/usr/local/opt/ncurses --with-ssl=/usr/local/opt/openssl@3 --with-sqlite=/usr/local/opt/sqlite

Compilation CFLAGS: -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -I/usr/local/opt/libgpg-error/include -I/usr/local/Cellar/libassuan/2.5.6/include -I/usr/local/Cellar/gpgme/1.22.0/include -I/usr/local/Cellar/neomutt/20231023/include -I/usr/local/include/lua -I/usr/local/Cellar/neomutt/20231023/include -I/usr/local/Cellar/neomutt/20231023/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/neomutt/20231023/include -DNCURSES_WIDECHAR -I/usr/local/Cellar/neomutt/20231023/include -I/usr/local/Cellar/openssl@3/3.1.3/include -I/usr/local/Cellar/libidn2/2.3.4_1/include -I/usr/local/Cellar/pcre2/10.42/include -I/usr/local/opt/lmdb/include -I/usr/local/Cellar/tokyo-cabinet/1.4.48/include -I/usr/local/Cellar/neomutt/20231023/include -O2

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  +autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo -gnutls +gpgme 
  -gsasl +gss +hcache -homespool +idn -inotify -locales_hack +lua -mixmaster 
  +nls +notmuch +openssl +pcre2 +pgp +sasl +smime +sqlite +sun_attachment 
  +truecolor 

MAILPATH="/var/mail"
PKGDATADIR="/usr/local/Cellar/neomutt/20231023/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/usr/local/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

### Operating System and its version

MacOS Monterey 12.7 x86_64

### Installation

Updates via Homebrew.

### Were you using multiple copies of NeoMutt at once?

No.

### Other information

For normal sending of email via the NeoMutt UI, i use `set sendmail="msmtp"`. That still works okay.

For this sole task, monthly email is automatically sent via crontab. Okay until this NeoMutt upgrade.

**Comment by flatcap** [comment]
Hmm... sorry about that.

A couple of us have tried to repeat the crash, without success: (Linux, BSD).

I tried **2023-10-06** and **2023-10-23** (both with [ASAN](https://neomutt.org/dev/analysis/asan) enabled).
I sent email from the GUI, the command line and the command line under cron (just in case).
All worked correctly.

It's very odd that **2023-10-06** works for you, but **2023-10-23** doesn't.
Very little code changed between the two versions,
and of the code that _did_ change, little would be seen when sending from the command line.

> I do not know about a 'coredump' file. Never tried to do such.

OK, I'll try to find some other Homebrew users to assist.

Meanwhile, please can you run with debugging enabled:
```
neomutt -d6 ....
```

That'll log a **lot** of info to `~/.neomuttdebug0`
Please can you post the end of that, here.

**Be careful** to remove any personal details.

Thanks!

**Comment by dcrossleyau** [comment]
Looking back, i see that the last time that my scheduled automated send happened was just prior to installing version 20231006, so it would have been working for me in 20230517. Sorry to mislead.

I will attend to the debug report tomorrow. Thanks for helping.

**Comment by dcrossleyau** [comment]
There is no debug file created when using the '-d' option, nor when also using the '-l' option for a debug file.

I tried without "batch mode" and it put me in the editor, where i could edit the message, and it was successfully sent. The debug file was then created.

**Comment by dcrossleyau** [comment]
Found the problem.

I have set of colour definitions that i have been happily using for ages ("solarized-dark-256" plus some others).

Started commenting-out one-by-one. This was the killer:

```
mono  body  bold  "^gpg: Good signature"
```

Would someone please edit this issue ticket title to clarify the problem.

Thanks for wonderful NeoMutt -- would be lost without it.

**Comment by flatcap** [comment]
Argh!  Now I understand.

A small bit of some upcoming colour changes crept into the release.
It caused `mono` commands to be run when they shouldn't have been.

I've nearly finished the colour code, then I'll make a new release next week.

> Thanks for wonderful NeoMutt

You're welcome!

**Comment by flatcap** [comment]
Colour fixes merged in f9c11d60d.
